components/suggestion/emoticon_provider.tsx

```diff
@@ -21,6 +23,19 @@ import Provider from './provider';
 export const MIN_EMOTICON_LENGTH = 2;
 export const EMOJI_CATEGORY_SUGGESTION_BLOCKLIST = ['skintone'];
 
+type ProviderResults = {
+    matchedPretext: string;
+    terms: string[];
+    items: Array<Record<string, any>>;
+    component?: React.ReactNode;
+};
```

Comment:
can we make this a generic type? This would get rid of the `any` in the `Record`:

```suggestion
type ProviderResults<T> = {
    matchedPretext: string;
    terms: string[];
    items: Array<Record<string, T>>;
    component?: React.ReactNode;
};
```

And as a second note we could move this to the root provider class so that every implementation of it can just import from there, but this is likely to be done in a separate PR @M-ZubairAhmed